addons/mail/static/src/messaging/component/attachment/attachment.js

```diff
@@ -119,6 +119,7 @@ class Attachment extends Component {
     _onClickUnlink(ev) {
         ev.stopPropagation();
         this.attachment.remove();
+        this.trigger('o-attachment-removed');
```

Comment:
Note that this work only because `attachment.remove()` is async and there is no `await`, as removal of attachment must destroy this component.

The correct solution is to handle change in entities and let components reflect on any changes. This comes down, again, to the "spying" stuff in the form_renderer that you mentioned the other day.

Also from a glance, this event is emitted and intercepted just to emit another event. That's quite a lot of message-passing communication in components, and I think you know that I don't really like this because this has many flaws. We need to use them as little as possible, mostly for integration with old code (widgets) or when we have comply with OWL components that are designed (IMO, badly) like this...

Reply:
I understand your points but I don't know how we could do "for now" as we need to now when an attachment is added/deleted but only in the context of the `AttachmentBox`. So IMO the logic should be in attachment box but as mentioned I have no tool to check this in another way without "spying" which I think should be well-designed before implementation.

Reply:
Ok, for now let's just keep in mind we will have to adapt this soon.